docs/YML-CBuild-Format.md

```diff
@@ -33,13 +33,87 @@ File                | Description
 
+The `*.cbuild-pack.yml` file has the following usage:
+
+- It contains strict list of software packs used including version information and ranges from `.csolution.yml` and all subsequent potentially included `.cproject.yml` and `.clayer.yml` files.
+- This file is used as `lock-file` to ensure that subsequent runs of `csolution` use the very same packs and pack versions on all computers / setups / runs / environments.
+
+### Pack locking
+
+An application contains a set of packs coming from different places, e.g. from the `csolution` directly, or indirectly from `cprojects` or `clayers`.
+In order to have consistent pack usage in the application, as well as allowing projects to evolve and add new `target-types` or `build-types` but still remain on the same shared pack versions, the `cbuild-pack.yml` is introduced.
+
+It works in the following way. An entire application has a set of `pack requirements`. These requirements can come from many different locations or contexts, and may be:
+- specified exactly, e.g. `ARM::CMSIS@5.9.0`
+- specified with range, e.g. `ARM::CMSIS@>=5.8.0`
+- specified without version, e.g. `ARM::CMSIS`
+- specified with wildcards on the pack name, e.g. `ARM::CMSI*`
+- specified without pack name, e.g. `ARM`
+
+All these `pack requirements` are `resolved` into exact versions in the `cbuild-pack.yml` file as a list of items on the following format:
+```yml
+cbuild-pack:
+  resolved-packs:
+    - resolved-pack: ARM::CMSIS@5.9.0
+      selected-by:
+        - ARM
+        - ARM::CMSI*
+        - ARM::CMSIS
+        - ARM::CMSIS@>=5.8.0
+        - ARM::CMSIS@5.9.0
+```
+
+If a context is added or changed, the `selected-by` is used to ensure that the `pack requirements` are resolved to a `consistent` pack version, reducing surprising versions being selected in the entire application.
+If a `pack requirement` is no longer present in the application, then the `cbuild-pack.yml` file will be *cleaned* from the relevant entries during the `cbuild-pack.yml`-generation step.
+
+The location of the `cbuild-pack.yml` file follows the `csolution.yml` file.
+
+* `convert` - `Uses` **and updates** cbuild-pack.yml
+* `list ...` - `Uses` cbuild-pack.yml
+* `run` - `Uses` cbuild-pack.yml
+* `update-rte` - `Uses` **and updates** cbuild-pack.yml
+
+```mermaid
+graph TD;
+  csolution.yml
+  cbuild-pack.yml
+  cproject.yml
+
+subgraph Loading
+  res1["<center><u>Resolving step 1 (PopulateContexts)</u></center><ul><li>Reading the files</li><li>Adds cbuild-pack.yml\npack requirements to the solution</li></ul>"]
+  res2["<center><u>Resolving step 2 (AddPackRequirements)</u></center><ul><li>Pack version ranges are matched to fully qualified versions using cbuild-pack.yml</li><li>Pack wildcards are matched to fully qualified versions using cbuild-pack.yml</li><li>Pack wildcards are kept for further expansion in the model</li><li>Local packs are left as-is</li></ul>"]
```

Comment:
What does "Pack wildcards are kept for further expansion in the model" mean?
After resolving the packs, no further processing should be needed.